applications/nrf_desktop/src/modules/dfu.c

```diff
@@ -337,6 +400,10 @@ static void handle_dfu_data(const uint8_t *data, size_t size)
 		return;
 	}
 
+	if (IS_ENABLED(CONFIG_DESKTOP_DFU_LOCK) && dfu_lock_claim(&config_channel_owner)) {
+		return;
+	}
```

Comment:
Maybe we could assert on claiming lock success? If flash area is open we should be sure that config channel DFU has already started and therefore we must be the DFU lock owner.